internal/providers/pke/pkeworkflow/create_subnet_activity.go

```diff
@@ -0,0 +1,170 @@
+	}
+}
+
+// CreateSubnetActivity responsible for setting up a Subnet for an EKS cluster
+type CreateSubnetActivity struct {
+	awsClientFactory *AWSClientFactory
+}
+
+// CreateSubnetActivityInput holds data needed for setting up
+// a Subnet for EKS cluster
+type CreateSubnetActivityInput struct {
+	AWSActivityInput
+	ClusterID   uint
+	ClusterName string
+	// the ID of the VPC to create the subnet into
+	VpcID string
+
+	// the ID of the Route Table to associate the Subnet with
+	RouteTableID string
+
+	// The CIDR of the subnet
+	Cidr string
+
+	// The availability zone of the subnet
+	AvailabilityZone string
+}
+
+// CreateSubnetActivityOutput holds the output data of the CreateSubnetActivity
+type CreateSubnetActivityOutput struct {
+	SubnetID         string
+	Cidr             string
+	AvailabilityZone string
+}
+
+func (a *CreateSubnetActivity) Execute(ctx context.Context, input CreateSubnetActivityInput) (*CreateSubnetActivityOutput, error) {
+	logger := activity.GetLogger(ctx).Sugar().With(
+		"organization", input.OrganizationID,
+		"cluster", input.ClusterName,
+		"vpcID", input.VpcID,
+		"routeTableID", input.RouteTableID,
+		"subnetCidr", input.Cidr,
+		"availabilityZone", input.AvailabilityZone,
+		"secret", input.SecretID,
+	)
+
+	client, err := a.awsClientFactory.New(input.OrganizationID, input.SecretID, input.Region)
+	if err = errors.WrapIf(err, "failed to create AWS session"); err != nil {
+		return nil, err
+	}
+
+	buf, err := ioutil.ReadFile("templates/pke/subnet.cf.yaml")
+	if err != nil {
+		return nil, errors.WrapIf(err, "loading CF template")
+	}
+
+	stackName := aws.String("pke-subnet-" + input.ClusterName + "-" + input.AvailabilityZone)
+
+	if input.Cidr != "" {
+		logger.Debug("creating subnet")
+
+		cloudformationClient := cloudformation.New(client)
+
+		stackParams := []*cloudformation.Parameter{
+			{
+				ParameterKey:   aws.String("VpcId"),
+				ParameterValue: aws.String(input.VpcID),
+			},
+			{
+				ParameterKey:   aws.String("RouteTableId"),
+				ParameterValue: aws.String(input.RouteTableID),
+			},
+			{
+				ParameterKey:   aws.String("SubnetBlock"),
+				ParameterValue: aws.String(input.Cidr),
+			},
+			{
+				ParameterKey:   aws.String("AvailabilityZoneName"),
+				ParameterValue: aws.String(input.AvailabilityZone),
+			},
+		}
+		createStackInput := &cloudformation.CreateStackInput{
+			DisableRollback:  aws.Bool(true),
+			StackName:        stackName,
+			Parameters:       stackParams,
+			Tags:             getSubnetStackTags(input.ClusterName),
+			TemplateBody:     aws.String(string(buf)),
+			TimeoutInMinutes: aws.Int64(10),
+		}
+
+		_, err := cloudformationClient.CreateStack(createStackInput)
+		if err != nil {
+			return nil, errors.WrapIf(err, "create stack failed")
+		}
+
+		describeStacksInput := &cloudformation.DescribeStacksInput{StackName: stackName}
+		err = eksWorkflow.WaitUntilStackCreateCompleteWithContext(cloudformationClient, ctx, describeStacksInput)
+
+		if err != nil {
```

Comment:
In case of error you can collect the stack events that error out with `NewAwsStackFailure` and return that.